packages/lib-classifier/dev/components/App/App.js

```diff
@@ -109,6 +122,7 @@ class App extends React.Component {
               onAddToCollection={(subjectId) => console.log(subjectId)}
               onCompleteClassification={(classification, subject) => console.log('onComplete', classification, subject)}
               project={this.state.project}
+              workflowID={this.workflowId()}
```

Comment:
This is a pretty minor non-blocking comment since this only affects our dev classifier setup, but I've only seen the pattern of calling a function in the prop definition of a component when you dynamically determining what to render and the return value is a component. I'm not sure if this is a good idea otherwise, since this will be called every time this mini app is re-rendered and short circuits the lifecycle methods and any performance benefits those have. I think it'd be safer to call this in `componentDidMount`? Regardless, this is a non-blocking question.

Reply:
Good catch. `workflowId` doesn't even need to be a class method. It could be run once, outside the component, and the workflow ID passed in as a prop to App.

Reply:
Yeah, I think we had talked about ages ago about the potential of there being a functional workflow selection library that could be used independently by any app and this is one of the strategy types.

Reply:
Yeah, that could happen outside the classifier, or the `selectWorkflow` action could invoke a strategy in order to choose a workflow, with some sort of default fallback. Outside the classifier could mean that it's the project's responsibility to choose a selection strategy, which could be useful for Gravity Spy.

If we made `workflowID` a required prop for the classifier, that might allow some simplification of the stores by removing some of the autorun code, and the default fallback. Do we want consuming apps to be responsible for selecting a workflow and passing it in? That's probably out of the scope of this PR.

I've moved the workflow parameter function out of the classifier component and into the dev app. Should I do the same with the code that gets the project from the URL? At the moment, you can do something like this with the production classifier but it seems like params should only be for development.
https://www.zooniverse.org/projects/nora-dot-eisner/planet-hunters-tess/classify?project=908&workflow=3370&env=staging

Reply:
Agreed that it's out of scope of this PR. Would you add notes to the RFC about workflow selection? I don't want this to get lost in this PR. Thanks! https://github.com/zooniverse/front-end-monorepo/issues/1132

We'll have to address workflow selection soon, so capturing an idea of what we want to do would be good.

Reply:
Will do. I think it's a variation on the idea here: https://github.com/zooniverse/front-end-monorepo/issues/1132#issuecomment-548516888